### PR TITLE
Ensure we don't keep trying forever to find new automatic sentinel

### PR DESCRIPTION
**Pull request by quinnj**

### CODE REVIEW

**Comment by bkamins** [comment]
Probably it would be good to tell the user what to do about this error in the docs :).

Also instead of
```
error("error trying to automatically find a new sentinel for SentinelArray")
```
I think it would be better to have a custom and exported exception, so that in users code you can cleanly `catch` it and e.g. fallback to array of `Union{Missing, T}`. The issue is that `error` produces too generic error I think (and in practice you probably want to catch this error in 100% of cases).

**Comment by quinnj** [comment]
Good points; pushed a bunch of updates that should cover everything we've talked about in #1 (and your comments here).

File: README.md [hunk 1]
```diff
@@ -7,21 +7,44 @@
 The package is registered in the [`General`](https://github.com/JuliaRegistries/General) registry and so can be installed at the REPL with `] add SentinelArrays`.
 
 ## Usage
-A `SentinelArray` should act/wrap a regular `Array` for any given operation.
 
-Examples
-```julia
-x = SentinelVector{Float64}(undef, 10)
+    SentinelArray(A::AbstractArray, sentinel, value)
+    SentinelVector{T}(undef, len, sentinel, value)
```
Comment (bkamins) [inline README.md:12]: also `SentinelMatrix` now - right?

Comment (quinnj) [reply README.md:12]: Yeah good point. I added it to docs and added a test as well.

**Pull request merged**